### PR TITLE
[HB-6882] Amazon Compliance Update and 9.8.9 version bump

### PR DESCRIPTION
**Pull request by bichenwang**

### CODE REVIEW

File: AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt [hunk 1]
```diff
@@ -391,93 +472,144 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
...
+                    AmazonPublisherServicesAdapterPreBidRequest(
+                        chartboostPlacement = request.chartboostPlacement,
+                        format = request.format,
+                        amazonSettings = amazonSettings,
+                        ccpaPrivacyString = ccpaPrivacyString,
+                    )
+                )?.fold({ adInfo ->
+                    CoroutineScope(Main.immediate).launch {
+                        placementToPreBidAdInfoMap[placement] = adInfo
+                    }
```
Comment (cb-jpadilla) [inline AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt:487]: We already are inside a Main scope, does `fold` change the scope? 🤔  I suppose, always good to be safe.

File: AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt [hunk 2]
```diff
@@ -185,14 +263,22 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
         )
             .trim()
             .takeIf { it.isNotEmpty() }?.let { appKey ->
-                AdRegistration.getInstance(appKey, context)
+                val useManagedPrebidding =
```
Comment (chauduyphanvu) [inline AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt:266]: `shouldUseManagedPrebidding`?

**Review by chauduyphanvu — APPROVED** [review]
LGTM. Just some nits. Also I'd run `ktlint --format` on this

**Review by cb-jpadilla — COMMENTED** [review]
Minor nits, else looks good to me.

**Comment by cb-jpadilla** [comment]
Oh @chauduyphanvu, had similar feedback. Let me remove my redundant ones.

File: AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt [hunk 3]
```diff
@@ -869,6 +992,7 @@ private class AdListener(
                     "Unable to fire onPartnerAdDismissed for Amazon Publisher Services adapter. Listener is null.",
                 )
             }
+            fullscreenAd = null
```
[the remark this answers is not in the document]
Comment (bichenwang) [reply AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt:1023]: no clue. this line doesn't even do anything anyway. not sure why I added it lol

**Review by cb-jpadilla — APPROVED** [review]
Lgtm

**Review by chauduyphanvu — APPROVED** [review]

**Pull request merged**